### DISCUSSION TITLE
How to find the clock domain of a sequential cell?

### DISCUSSION THREAD

**Discussion by fgaray** [The-OpenROAD-Project/OpenROAD#3013] [Q&A] [answered]
Hi OpenRoad devs,

Part of the scan architect step (to create the scan chains) is to see what clock is driving each one of the cells of the design. I was wondering: what is the best way in OpenRoad to find out what is the clock driver and the edge (Rising or Falling) of a given cell?.

I can always traverse back the nets from a clock pin but maybe there is already some infra that I can use for this.

Thanks!,
Felipe

**Comment by maliberty** [comment, marked as answer]
I think Sta::clocks(const Pin *pin) is what you want.

**Reply by fgaray** [reply]
Thanks Matt, that worked perfectly!
